src/app/models/instance/InstanceMetagame.ts

```diff
@@ -0,0 +1,71 @@
+import { Column, ObjectIdColumn, Entity, Index, ObjectID } from "typeorm";
+import { World, worldArray } from '../../constants/world';
+import { Zone, zoneArray } from '../../constants/zone';
+import { MetagameEventType, metagameEventTypeArray } from '../../constants/metagameEventType';
+import { Ps2alertsEventState, ps2alertsEventStateArray } from '../../constants/ps2alertsEventState';
+
+@Entity()
+@Index(["world", "censusInstanceId"], { unique: true })
+export class InstanceMetagame {
+
+    @ObjectIdColumn()
+    _id: ObjectID;
+
+    @Column({
+        type: "string",
+        nullable: false
+    })
+    instanceId: string;
+
+    @Column({
+        type: "enum",
+        enum: worldArray,
+        nullable: false
+    })
+    world: World;
+
+    @Column({
+        type: "date",
+        nullable: false
+    })
+    timeStarted: Date;
+
+    @Column({
+        type: "date",
+        nullable: true
+    })
+    timeEnded: Date;
```

Comment:
Does this not also need `| null`?

Reply:
Figured the nullable:true does the same. Also wasn't sure the Column decorator supported multiple types

Reply:
Property key should have question mark.